prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,251 @@
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040, 2050.
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int kwhmi: fuel efficiency, should vary based on vehicle type and model_year.
+    :param int power: charger power, EVSE kW.
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5;
+        1-home only, 2-home and work related, 3-anywhere if possibile,
+        4-home and school only, 5-home and work and school.
+    :param str veh_type: determine which category (LDV or LDT) to produce charging profiles for
+    :param str filepath: the path to the NHTS mat file.
+    :param int trip_strategy: determine to charge after any trip (1) or only after the last trip (2)
+    :return: (*numpy.ndarray*) -- charging profiles.
+    """
+
+    # load NHTS data from function
+    if veh_type.lower() == "ldv":
+        newdata = data_helper.remove_ldt(data_helper.load_data(census_region, filepath))
+
+    elif veh_type.lower() == "ldt":
+        newdata = data_helper.remove_ldv(data_helper.load_data(census_region, filepath))
+
+    # updates the weekend and weekday values in the nhts data
+    newdata = data_helper.update_if_weekend(newdata)
+
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata = newdata.reindex(
+        list(newdata.columns)
+        + [
+            "trip start battery charge",
+            "trip end battery charge",
+            "charging power",
+            "charging time",
+            "charging consumption",
+            "BEV could be used",
+            "trip number",
+        ],
+        axis="columns",
+        fill_value=0,
+    )
+
+    input_day = data_helper.get_input_day(data_helper.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = data_helper.get_data_day(newdata)
+
+    weekday_electricload = get_electricload(
+        newdata, data_day, 2, veh_range, kwhmi, power, location_strategy, trip_strategy
+    )
+    weekday_output_electricload = get_output_electricload(weekday_electricload)
+
+    weekend_electricload = get_electricload(
+        newdata, data_day, 1, veh_range, kwhmi, power, location_strategy, trip_strategy
+    )
+    weekend_output_electricload = get_output_electricload(weekend_electricload)
+
+    outputs = [weekend_output_electricload, weekday_output_electricload]
+
+    for day_iter in range(len(input_day)):
+        # weekday = 1 if weekday, 0 if not weekday
+        weekday = input_day[day_iter] - 1
+
+        if day_iter == len(input_day) - 1:
+            TRANS_charge[day_iter * 24 :] += outputs[weekday][:24].transpose()
+            TRANS_charge[:24] += outputs[weekday][24:48].transpose()
+
+        else:
+            TRANS_charge[day_iter * 24 : day_iter * 24 + 48] += outputs[
+                weekday
+            ].transpose()
```

Comment:
Do we need to transpose at all here? We're adding one 1D array to another 1D array.